### ISSUE TITLE
Add missing kernels from GPML

### ISSUE THREAD

**Issue by sharanry** [theogf/KernelFunctions.jl#44]

**Comment by theogf** [comment]
Do you have a list somewhere?

**Comment by sharanry** [comment]
From a quick look, these seem to be the missing kernels


- covCos
- covDiscrete
- covEye
- covFBM
- covGabor
- covGaborard
- covGaboriso
- covLINard
- covLINiso
- covLINone
- covMaha
- covMask
- covMaternard
- covMaterniso
- covNNone
- covOU
- covPER
- covPERard
- covPERiso
- covPP
- covPPard
- covPPiso
- covPeriodic
- covPeriodicNoDC
- covPref
- covRQard
- covRQiso
- covSEard
- covSEiso
- covSEisoU
- covSEproj
- covSEvlen
- covSM
- covULL
- covW
- covWarp

**Comment by sharanry** [comment]
@theogf @willtebbutt Do you suggest I make separate PRs for each kernel or just one for all of them combined?

**Comment by theogf** [comment]
A lot of this kernel are actually already implemented `covSEard` etc. I think having one PR per kernel seems fine. Maybe you can start by opening an issue for the ones missing so anyone can participate?

**Comment by willtebbutt** [comment]
Yeah, you really just need to implement the "base" kernel -- the input transformations provided by `KernelFunctions` gives you the rest for free (see the `@kernel` macro in #38 and more generally the various `Transform`s).

I think maybe `covSEvlen` might need a special implementation, but the various `---ard`, `---iso`, `---proj` etc should be fine.

I agree that we don't have the `Cosine`, `Gabor`, `Eye`, `FBM`, `Maha`, `NN`, `PP`, `Pref`, `SM`, `ULL`, or `W` (wiener process). (I'm not even sure what some of these are lol)

There are also more recent kernels that GPML doesn't have, but we might want to implement. I'm thinking the various DNN kernels that have cropped up in the literature in recent years (I can't remember their names off the top of my head though)

**Comment by theogf** [comment]
Haha same for me it's the first time I hear about some of those.
I will try to write some documentation on how to make custom kernels. But you can already have a look at the beginning of  https://github.com/theogf/KernelFunctions.jl/blob/master/test/test_custom.jl

**Comment by sharanry** [comment]
Yeah. The function names aren't the most helpful. I have added a short description to them. Hope that helps.

**Comment by willtebbutt** [comment]
Very excited for having implementations of these. I've had [an issue](https://github.com/willtebbutt/Stheno.jl/issues/50) on Stheno.jl about this for a while, so it would be amazing to have this stuff done here!

**Comment by sharanry** [comment]
@willtebbutt @theogf I am trying to implement [Fractional Brownian motion covariance ](https://github.com/alshedivat/gpml/blob/master/cov/covFBM.m) given by
```
    κ(x,y) =  ( |x|²ʰ + |z|²ʰ - |x-z|²ʰ ) / 2
``` 
 with Hurst index `h` from (0,1).

I don't think the current structure of `kappa` and `metric` can be used directly implement this. I am considering implementing a modified `kappa` function to serve such cases. Any thoughts on this?
